### DISCUSSION TITLE
Secret scanning validity checks enablement through API

### DISCUSSION THREAD

**Discussion by Jegan-Kunniya** [community/community#109535] [Code Security] [answered]
### Select Topic Area

Question

### Body

Hello there,

With the validity checks for secrets introduced, and is being available for selected/limited partners, I am looking to enable that capability on all the existing GHAS repos.

Is there an API to do so? I looked at the REST API documentation and I didn't find one.

if there is no API, what would suggest automating this scenario?

Thanks

**Comment by aashah** [comment, marked as answer]
Yeahp, under https://docs.github.com/en/enterprise-cloud@latest/rest/repos/repos?apiVersion=2022-11-28#update-a-repository there is a key called `secret_scanning_validity_checks`.

This also exists at the org + business level.

**Reply by Jegan-Kunniya** [reply]
Thank you @aashah for your input. I could update my automation with the `secret_scanning_validity_checks` property to enable it for all onboarded GHAS repos.
